discovery/manager_test.go

```diff
@@ -949,6 +951,91 @@ scrape_configs:
 	}
 }
 
+func TestGaugeFailedConfigs(t *testing.T) {
+	var (
+		fcGauge prometheus.Gauge
+		err     error
+	)
+
+	cfgOneText := `
+scrape_configs:
+- job_name: prometheus
+  consul_sd_configs:
+  - server: "foo:8500"
+    tls_config:
+      cert_file: "/tmp/non_existent"
+  - server: "bar:8500"
+    tls_config:
+      cert_file: "/tmp/non_existent"
+  - server: "foo2:8500"
+    tls_config:
+      cert_file: "/tmp/non_existent"
+`
+	cfgOne := &config.Config{}
+
+	err = yaml.UnmarshalStrict([]byte(cfgOneText), cfgOne)
+	if err != nil {
+		t.Fatalf("Unable to load YAML config cfgOne: %s", err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	discoveryManager := NewManager(ctx, log.NewNopLogger())
+	discoveryManager.updatert = 100 * time.Millisecond
+	go discoveryManager.Run()
+
+	c := make(map[string]sd_config.ServiceDiscoveryConfig)
+	for _, v := range cfgOne.ScrapeConfigs {
```

Comment:
You can just create the `map[string]ServiceDiscoveryConfig` instead of unmarshalling text.

Reply:
I will do that with some other tests during refactor.